rpcserver.go

```diff
@@ -2939,6 +2939,43 @@ func handlePing(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (inter
 	return nil, nil
 }
 
+// getMempoolTxsForAddressRange looks up and returns all transactions from the
+// mempool related to the given address. The, `limit` parameter
+// should be the max number of transactions to be returned. Additionally, if the
+// caller wishes to seek forward in the results some amount, the 'seek'
+// represents how many results to skip.
+// It will return the array of fetched transactions, along with the amount
+// of transactions that were actually skipped.
+func getMempoolTxsForAddressRange(s *rpcServer, addr btcutil.Address, skip int,
+	limit int) ([]*database.TxListReply, int, error) {
+
+	memPoolTxs, err := s.server.txMemPool.FilterTransactionsByAddress(addr)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// If we're asked to skip more transactions than we have,
+	// we skip them all and return an empty slice.
+	if skip >= len(memPoolTxs) {
+		return nil, len(memPoolTxs), nil
+	}
+
+	var result []*database.TxListReply
+
+	// Otherwise, calculate the range we have to return and return it.
+	rangeEnd := skip + limit
+	if rangeEnd > len(memPoolTxs) {
+		rangeEnd = len(memPoolTxs)
+	}
+
+	for _, tx := range memPoolTxs[skip:rangeEnd] {
```

Comment:
So, for the bits of the `addrindex` stored in the DB, `skip` allows one to move forward in terms of chronological/dependency order since the `blkHeight` and `txoffset` portions of the index are stored in `BigEndian` order. 

However, the way `skip` is currently implemented within the `mempool` (was this way before your PR), returned entries are always randomized (Go's map iteration is random). If it enhances your use case, it may be worth it to also define some sort of ordering for txs returned from the mempool. A possible order (suggested in the [commit which added ordering](https://github.com/aakselrod/btcd/commit/ead39153aff20d191e4d28c39b86e82cce0924d6)), would arise if one did a topological sort of the mempool txns. If implemented, this would result in txns being returned in dependency order.